tools/distpartitioning/parmetis_postprocess.py

```diff
@@ -0,0 +1,99 @@
+import os
+import sys
+import constants
+import argparse
+import pyarrow
+import pyarrow.csv as csv
+import logging
+import platform
+import numpy as np
+
+from pathlib import Path
+from utils import read_json, get_node_types, get_idranges
+
+def post_process(params):
+    """Auxiliary function to read the parmetis output file and generate
+    metis partition-id files, sorted, per node-type. These files are used
+    by the dist. graph partitioning pipeline for further processing.
+
+    Parameters:
+    -----------
+    params : argparser object
+        argparser object to capture command line options passed to the 
+        executable
+    """
+    logging.info("Starting to process parmetis output.")
+    
+    logging.info(params.schema_file)
+    logging.info(params.parmetis_output_file)
+    assert os.path.isfile(params.schema_file)
+    assert os.path.isfile(params.parmetis_output_file)
+    schema = read_json(params.schema_file)
+
+    metis_df = csv.read_csv(params.parmetis_output_file, 
+        read_options=pyarrow.csv.ReadOptions(autogenerate_column_names=True),
+        parse_options=pyarrow.csv.ParseOptions(delimiter=' '))
+    global_nids = metis_df['f0'].to_numpy()
+    partition_ids = metis_df['f1'].to_numpy()
+
+    sort_idx = np.argsort(global_nids)
+    global_nids = global_nids[sort_idx]
+    partition_ids = partition_ids[sort_idx]
+
+    ntypes_ntypeid_map, ntypes, ntid_ntype_map = get_node_types(schema)
+    type_nid_dict, ntype_gnid_offset = \
```

Comment:
don't do this '\' manually, black will take care of it for you.

Reply:
Agreed.